spring-tx/src/main/java/org/springframework/transaction/jta/JtaTransactionManager.java

```diff
@@ -547,8 +547,8 @@ protected void initTransactionSynchronizationRegistry() {
 	 * @return a corresponding UserTransaction handle
 	 */
 	protected UserTransaction buildUserTransaction(TransactionManager transactionManager) {
-		if (transactionManager instanceof UserTransaction) {
-			return (UserTransaction) transactionManager;
+		if (transactionManager instanceof UserTransaction userTransaction) {
```

Comment:
This fails the build due to a Checkstyle limitation that prevents Checkstyle from differentiating between the location `userTransaction` and `this.userTransaction`.